### PR TITLE
Validate that apparmor_parser binary is available on the host

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
If AppArmor is enabled, then we now pre-check on CRI-O startup if the
`apparmor_parser` binary is either in `/sbin` or in `$PATH`. This way we
can let the user know that there is still a missing runtime dependency.
#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None
#### Does this PR introduce a user-facing change?

```release-note
- Added validation of the existence of the `apparmor_parser` binary (either in `/sbin` or in `$PATH`) on CRI-O start, if the feature is enabled by the build or the host.
```

### CODE REVIEW

**Comment by haircommander** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by kolyshkin** [comment]
IMHO this check should be added to github.com/containers/libpod/pkg/apparmor.IsEnabled() instead, since it is the one that uses and needs that binary (and cri-o does not).

It did that before by calling runc's libcontainer/apparmor.IsEnabled(), but now runc doesn't check binary presense (as it's not using it itself).

File: internal/config/apparmor/apparmor.go [hunk 1]
```diff
@@ -38,9 +37,9 @@ func (c *Config) LoadProfile(profile string) error {
 		return nil
 	}
 
-	if profile == unconfined {
+	if profile == v1.AppArmorBetaProfileNameUnconfined {
 		logrus.Info("AppArmor profile is unconfined which basically disables it")
-		c.defaultProfile = unconfined
+		c.defaultProfile = v1.AppArmorBetaProfileNameUnconfined
```
Comment (kolyshkin) [inline internal/config/apparmor/apparmor.go:42]: This should be a separate commit, please.

**Comment by saschagrunert** [comment]
> IMHO this check should be added to github.com/containers/libpod/pkg/apparmor.IsEnabled() instead, since it is the one that uses and needs that binary (and cri-o does not).

The most recent version of Podman does use containers/common instead of its own implementation. We probably have to add it there and switch over to the new library. I really like to make the issue (binary is missing) visible to the user instead of just disabling AppArmor. It is also a matter _when_ to report the issue to the user. Having the validation on CRI-O startup seems more reasonable to me than doing it late on container creation time.

I'll add the check to containers/common and follow-up with this PR later on.

File: internal/config/apparmor/apparmor.go (hunk 1 above)
Comment (saschagrunert) [reply internal/config/apparmor/apparmor.go:42]: Will be a dedicated PR, thank you for the hint.

Comment (saschagrunert) [reply internal/config/apparmor/apparmor.go:42]: See https://github.com/cri-o/cri-o/pull/4098

**Comment by saschagrunert** [comment]
Following-up in https://github.com/containers/common/pull/269
